client/post-editor/index.js

```diff
@@ -14,6 +14,7 @@ module.exports = function() {
 	page( '/post', controller.pressThis, sitesController.siteSelection, sitesController.sites );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page( '/post/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.post );
+	page( '/post/revision/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.revision );
```

Comment:
From `getSiteFragment` code that you linked:

> There are 2 URL positions where we should look for the site fragment:  last (most sections) and second-to-last (post ID is last in editor)

Aren't we just adding a 3rd position here?  So, the only changes to `getSiteFragment` needed to support `/post/:site/:post/revisions` should be as follows:

- change `i = 2` to `i = 3`
- add `pieces.length` bounds checking, since otherwise we'll try to evaluate `pieces[ -1 ]`
- update the comments and the tests accordingly

> Last solution could be to attack the root cause and get `getSiteFragment` to use page.js's params first and then default to exploding the path ... there could be other weird usages out there so this feels like a risky move tbh.

yes, I also think this could easily break something, whether now or in the future.